src/Routing/RouteCollection.php

```diff
@@ -144,7 +144,11 @@ public function parse($url, $method = '')
                 return $r;
             }
         }
-        throw new MissingRouteException(['url' => $url]);
+        throw new MissingRouteException([
+            'method' => $method,
+            'url' => $url,
+            'message' => 'A "%s" route matching "%s" could not be found.',
```

Comment:
Yes, that certainly makes sense to me. I've updated the branch with a version that does that.